README.md

```diff
@@ -124,6 +125,20 @@ Key:
 * `=` Maintenance is manual, only severe security bugs will be patched.
 * `-` Deprecated; please upgrade.
 
+#### Dependency Matrix
+Since there are multiple libraries that client-go is either required to, or can work
+with, this is a list of all the libraries and versions that work with each other.
```

Comment:
I would prefer not to make this in the form of a chart. I would say, "here are the dependency libraries you might need: `<list>`; corresponding commits are all tagged with `kubernetes-x.x.x` tags. If you use godep or glide you should not need to know this."

I would *especially* avoid mentioning e.g. `release-1.6` branch in the apimachinery repo, as IMO it is not versioned yet and I don't want people to expect that pattern to continue--it should switch to a semver system like client-go.

Reply:
But we need to give people an approach that works now, and for older (1.7 and lower) releases, the release branch is the only thing that exists in apimachinery.

Reply:
People starting fresh shouldn't use the older releases. People not starting fresh already have something that works and don't have this problem.

Reply:
Sorry, I don't understand this logic - are we assuming that there are no <= 1.7 kubernetes clusters that want to use client-go in the future?

Reply:
Newer client releases are backwards compatible, so basically, yes, I don't see why you would start out using something not the latest.

Reply:
Fair enough - then the simplification we've discussed further down will make more sense.